test/saithrift/tests/sail3.py

```diff
@@ -3450,3 +3450,87 @@ def verifyEcmpPktDistributionIpv6(self,max_itrs,ports,member):
                                       "Not all paths are equally balanced, %s" % count)
             print "All paths are  balanced with %s members" % member
 
+@group('l3')
+class L3LpbkSubnetTest(sai_base_test.ThriftInterfaceDataPlane):
+    def runTest(self):
+        """
+        Description:
+        Send traffic on a layer 3 router interface destined to an IPv4 address within
+        the same subnet. Have the nhop entry learnt on the same interface as incoming
+        interface. Verify that packet is send out on the same outgoing interface as
+        the incoming interface. Set the SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION
+        attribute to drop and verify that packet is not send out on the same outgoing
+        RIF as incoming RIF.
+
+        Steps:
+        Part 1:
+          1. Create Virtual Router V1 and enable v4 and v6.
+          2. Create a virtual router interface (rif_id1) and set the interface type as "PORT" for port1.
+          3. Create IPv4 neighbor entry (10.10.10.1) with MAC1 and associate with "RIF id 1".
+          4. Create next hop entry as rif_id1 (Loopback).
+          5. Send traffic on router interface rif_id1.
+          6. Verify the packet on port1.
+          7. Packet should received at port1.
+        Part 2:
+          1. Set router interface rif_id1 attribute as SAI_ROUTER_INTERFACE_ATTR_LOOPBACK_PACKET_ACTION
+          2. set the action as drop.
+          3. Send traffic on router interface rif_id1.
+          4. Verify the packet on port1.
+          5. Packet should not received at port1.
+
+        Clean up by remove the nhop, neighbor, route and the router interface.
+        """
+
+        switch_init(self.client)
+        port1 = port_list[0]
+        v4_enabled = 1
+        v6_enabled = 1
+        mac = ''
+
+        vr_id = sai_thrift_create_virtual_router(self.client, v4_enabled, v6_enabled)
+
+        rif_id1 = sai_thrift_create_router_interface(self.client, vr_id, SAI_ROUTER_INTERFACE_TYPE_PORT, port1, 0, v4_enabled, v6_enabled, mac)
+
+        addr_family = SAI_IP_ADDR_FAMILY_IPV4
+        ip_addr1 = '10.10.10.1'
+        ip_addr1_subnet = '10.10.10.0'
+        ip_mask1 = '255.255.255.0'
+        dmac1 = '00:11:22:33:44:55'
+        sai_thrift_create_neighbor(self.client, addr_family, rif_id1, ip_addr1, dmac1)
+        nhop1 = sai_thrift_create_nhop(self.client, addr_family, ip_addr1, rif_id1)
+        sai_thrift_create_route(self.client, vr_id, addr_family, ip_addr1_subnet, ip_mask1, nhop1)
+
```

Comment:
Test is passing after adding local route 10.10.0.0/16 to rif_id1

Reply:
Added local route